### ISSUE TITLE
Can't reach /api/v1/accessories/:id/checkout

### ISSUE THREAD

**Issue by markbrule** [grokability/snipe-it#9342]
#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode 
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**
I'm trying to check out an accessory via the API. I'm able to use the /api/v1/accessories/:id to see the accessory, and the /api/v1/accessories/:id/checkedout endpoint to see who already has checked out some of these accessories. When I try to POST to the /api/v1/accessories/:id/checkout endpoint, I get an http/404 error.

**To Reproduce**
snipeitdev is our local name for my dev SnipeIT instance
I have verified 1 is a valid Accessory ID, and 5 is a valid user ID

curl --request POST --url  http://snipeitdev/api/v1/accessories/1/checkout --header "Authorization: Bearer $token"  --header "Content-Type: application/json" --header "Accept: application/json" --data '{"assigned_to": 5}'

response:
{
  "payload": null,
  "messages": "404 endpoint not found",
  "status": "error"
}

**Expected behavior**
I expected one accessory to be checked out

**Server (please complete the following information):**
 - Snipe-IT Version v4.2.0 build 3497 (g590938f)
 - OS: Ubuntu 14.04
 - Web Server: apache
 - PHP Version  7.0.3

**Desktop (please complete the following information):**
 - OS: bash on the ubuntu 

**Error Messages**
Nothing comes out in the logs

**Additional context**
Everything else is working fine.
Checked the routes in the routes/web folder, all seem to be there.
Haven't made any changes to the base source.

Please do not post an issue without answering the related questions above. If you have opened a different issue and already answered these questions, answer them again, once for every ticket. It will be next to impossible for us to help you.

**Comment by boring10** [comment]
My guess is that, based on your version, that the API endpoint isn't available. Here is a link to the API endpoints for version 4.9.5 and it isn't listed as an option.

https://snipe-it.readme.io/v4.9.5/reference#accessories

<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/v4.9.5/reference#accessories">Overview</a></strong></div><div>Snipe-IT is a free, open source IT asset management system. Features include management of assets, users, licenses, accessories, consumables and components, as well as two-factor authentication, LDAP/AD syncing, and asset acceptance confirmation.</div></blockquote>

**Comment by markbrule** [comment]
Interseting; I am able to use /api/v1/accessories/:id/checkedout which isn't listed in the ref doc either. In routes/api.php I see:

    Route::group(['prefix' => 'accessories'], function () {
        Route::get('{accessory}/checkedout',
            [
                'as' => 'api.accessories.checkedout',
                'uses' => 'AccessoriesController@checkedout'
            ]
        );

And in routes/web/accessories.php I see:

Route::group([ 'prefix' => 'accessories', 'middleware' => ['auth']], function () {
    Route::get(
        '{accessoryID}/checkout',
        [ 'as' => 'checkout/accessory', 'uses' => 'AccessoriesController@getCheckout' ]
    );
    Route::post(
        '{accessoryID}/checkout',
        [ 'as' => 'checkout/accessory', 'uses' => 'AccessoriesController@postCheckout' ]
    );
    Route::get(
        '{accessoryID}/checkin/{backto?}',
        [ 'as' => 'checkin/accessory', 'uses' => 'AccessoriesController@getCheckin' ]
    );
    Route::post(
        '{accessoryID}/checkin/{backto?}',
        [ 'as' => 'checkin/accessory', 'uses' => 'AccessoriesController@postCheckin' ]
    );

});

It's as if the web/accessories.php aren't loaded. I'll try spinning up a more recent version and see if it works. Thanks for the pointer.

**Comment by boring10** [comment]
In the version 4.2.0 API routes you have

```php
/*--- Accessories API ---*/

    Route::resource('accessories', 'AccessoriesController',
        ['names' =>
            [
                'index' => 'api.accessories.index',
                'show' => 'api.accessories.show',
                'update' => 'api.accessories.update',
                'store' => 'api.accessories.store',
                'destroy' => 'api.accessories.destroy'
            ],
            'except' => ['create', 'edit'],
            'parameters' => ['accessory' => 'accessory_id']
        ]
    ); // Accessories resource

    Route::group(['prefix' => 'accessories'], function () {

        Route::get('{accessory}/checkedout',
            [
                'as' => 'api.accessories.checkedout',
                'uses' => 'AccessoriesController@checkedout'
            ]
        );
    }); // Accessories group
```

So you are correct that the checkedout is available even though it is not listed in the reference documentation but what you are wanting to do is not available in that version for the API.

What you are referencing is the web API which, I believe, is only for use from Snipe-IT itself (or an application that calls the tries to insert data through the website). What you want to look at is the <https://github.com/snipe/snipe-it/blob/846613c2448d8c640925feae606f0efa76c83705/routes/api.php> (<--version 4.2.0 for the api.php file).

<blockquote><img src="https://repository-images.githubusercontent.com/14346979/164a6580-7d79-11e9-865f-eae87593e11a" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/snipe/snipe-it">snipe/snipe-it</a></strong></div><div>A free open source IT asset/license management system - snipe/snipe-it</div></blockquote>

**Comment by markbrule** [comment]
This was resolved by upgrading to the most recent (5.1.4) release.
